libs/constrained-generators/src/Constrained/Test.hs

```diff
@@ -125,45 +137,53 @@ tests =
     , testSpec "maybeSpec" maybeSpec
     , testSpec "eitherSetSpec" eitherSetSpec
     , testSpec "fooSpec" fooSpec
-    , testSpec "intSpec" intSpec
+    , -- TODO: this spec needs double shrinking to shrink properly
+      -- so we need to figure something out about double-shrinking
+      testSpecNoShrink "intSpec" intSpec
```

Comment:
What is double shrinking?

Reply:
Think about it like this, you have a bug that only materialises when the function under test is given (x, y) such that x == y. If you try to shrink that you won't get far because shrinking of pairs works one component at a time - the property will fail for some (x, x) but shrink to (x', x) (to avoid an exploaion of shrinks paira tyoically shrink one component at a time). With double shrinking you'll try to shrink twice before concluding you're in a local minimum, so you'll try things like (x'', x) and crucially (x', x').